### ISSUE TITLE
getting KV no longer respects convert_to_hash

### ISSUE THREAD

**Issue by EugenMayer** [WeAreFarmGeek/diplomat#152]
related to #120 (introduced by this in 2.0.1) .. probably related to the concerns here https://github.com/WeAreFarmGeek/diplomat/issues/119#issuecomment-272527850

**Comment by EugenMayer** [comment]
very urgent for me, more or less fork or force, my complete stack is offline due to this ;/

**Comment by johnhamelink** [comment]
@eugenmayor obviously I’d recommend in the first instance you revert to an older version of the gem which fixes your immediate problems, then we can look at fixing. A really useful thing would be for you to create a unit test which mimics your current problems so we can fix them specifically

**Comment by EugenMayer** [comment]
Iam on vacation right now, won't be able to provide further details before tomorrow 11am CET. Basically all recursive convert to hash are now flatted arrays with keys as string, full path. So no more multi level hasmap but rather flatted array as it has been before convert to hash

**Comment by markitoxs** [comment]
Just to add more data to this:

```
Diplomat::Kv.get('g/web/maxconn')
=> "20000"
```

2.0.1 returns:
```
Diplomat::Kv.get('g/web/maxconn')
=> [{:key=>"g/web/maxconn", :value=>"20000"}]
```

**Comment by taharah** [comment]
Alright I'm going to revert the change in 2.0.1 that introduced this issue since it seems to have also introduced #153. We'll need to revisit the possibility of consistent returns at a later date.

**Comment by taharah** [comment]
This should be fixed in v2.0.2 of the gem @EugenMayer that reverts the change introduced in #120.

**Comment by taharah** [comment]
Closing this out since there has not been any additional reports of issues since reverting the change.
